tools/pydantic_codegen/plugin.py

```diff
@@ -0,0 +1,387 @@
+"""Defines a custom ariadne-codegen plugin to control Python code generation from GraphQL definitions.
+
+For more info, see:
+- https://github.com/mirumee/ariadne-codegen/blob/main/PLUGINS.md
+- https://github.com/mirumee/ariadne-codegen/blob/main/ariadne_codegen/plugins/base.py
+"""
+
+from __future__ import annotations
+
+import ast
+from collections import deque
+from contextlib import suppress
+from textwrap import dedent
+from typing import Any, Final, Iterable, TypeGuard, cast
+
+from ariadne_codegen import Plugin
+from ariadne_codegen.client_generators.constants import BASE_MODEL_CLASS_NAME
+from ariadne_codegen.codegen import generate_import, generate_import_from
+from graphlib import TopologicalSorter  # noqa # Run this only with python 3.9+
+from graphql import GraphQLSchema
+from pydantic import BaseModel
+
+DEFAULT_BASE: Final[str] = BASE_MODEL_CLASS_NAME
+GQL_BASE: Final[str] = "GQLBase"
+TYPENAME: Final[str] = "Typename"
+
+#: AST statements for...
+# `from __future__ import annotations`
+IMPORT_ANNOTATIONS = generate_import_from(["annotations"], from_="__future__")
+# `from .base import GQLBase, Typename, etc.`
+IMPORT_CUSTOM_TYPES = generate_import_from([GQL_BASE, TYPENAME], from_="base", level=1)
+# `import sys`
+IMPORT_SYS = generate_import(["sys"])
+
+
+#: Names that must be conditionally imported from `typing` or `typing_extensions` depending on python version.
+TYPING_IMPORTS_TO_REWRITE: Final[tuple[str, ...]] = (
+    "override",
+    "Annotated",
+    "Literal",
+)
+
+
+class FixFragmentOrder(Plugin):
+    """Codegen plugin to fix inconsistent ordering of fragments module.
+
+    HACK: At the time of implementation, the fragments module has inconsistent ordering of
+    - class definitions
+    - `Class.model_rebuild()` statements
+
+    See: https://github.com/mirumee/ariadne-codegen/issues/315. This plugin is a workaround in the meantime.
+    """
```

Comment:
**note:** This plugin a deliberate hack/workaround for an open issue with `ariadne-codegen` (see docstring)